inc/config.class.php

```diff
@@ -2512,13 +2512,19 @@ static function checkWriteAccessToDirs($fordebug = false) {
       ]);
 
       /* TODO: could be improved, only default vhost checked */
-      if ($fic = fopen('http://localhost'.$CFG_GLPI['root_doc'].'/index.php?skipCheckWriteAccessToDirs=1', 'r', false, $context)) {
+      $protocol = 'http';
+      if (isset($_SERVER['HTTPS'])) {
+         $protocol = 'https';
+      }
+      $uri = $protocol . '://' . $_SERVER['SERVER_NAME'] . $CFG_GLPI['root_doc'];
```

Comment:
indeed using server_name may be required for vhost/ssl, but mays be different from the client side (where is it retrieved) and from the server side.... btw no perfect solution... will need a JS check from the client

Reply:
Yes, this is not perfect; but maybe a bit better than just test localhost without https ;)